File: example/cs_ls/Config/Text.cs
namespace Config;

public partial class Text
{
    public required string Zh_cn { get; init; }
    public required string En { get; init; }
    public required string Tw { get; init; }
    private Text() {}

    public override string ToString()
    {
        return "(" + Zh_cn + "," + En + "," + Tw + ")";
    }

    internal static Text _create(Config.Stream os)
    {
        string[] texts = os.ReadTextsInPool();
        return new Text
        {
        Zh_cn = texts[0],
        En = texts[1],
        Tw = texts[2],
        };
    }
}

File: example/go_ls/config/Text.go
package config

type Text struct {
    Zh_cn string
    En string
    Tw string
}

func createText(stream *Stream) *Text {
    self := &Text{}
    texts := stream.ReadTextsInPool()
    if 0 < len(texts) {
        self.Zh_cn = texts[0]
    }
    if 1 < len(texts) {
        self.En = texts[1]
    }
    if 2 < len(texts) {
        self.Tw = texts[2]
    }
    return self
}

func (t *Text) String() string {
    result := ""
    result += t.Zh_cn
    result += ","
    result += t.En
    result += ","
    result += t.Tw
    return result
}
